### PR TITLE
[SandboxIR] Add missing VectorType functions

### PR DESCRIPTION
**Pull request by Sterling-Augustine**
Fills in many missing functions from VectorType

### CODE REVIEW

File: llvm/include/llvm/SandboxIR/Type.h [hunk 1]
```diff
@@ -317,7 +317,32 @@ class StructType : public Type {
...
+  Type *getElementType() const;
+
+  static VectorType *get(Type *ElementType, const VectorType *Other) {
+    return VectorType::get(ElementType, Other->getElementCount());
+  }
+
+  inline ElementCount getElementCount() const {
+    return cast<llvm::VectorType>(LLVMTy)->getElementCount();
+  }
+  static VectorType *getInteger(Context &Ctx, VectorType *VTy);
```
Comment (vporpo) [inline llvm/include/llvm/SandboxIR/Type.h:335]: We don't need the `Ctx` argument, you can get it like: `VTy->getContext()`. Same for the rest of the functions.

File: llvm/unittests/SandboxIR/TypesTest.cpp [hunk 2]
```diff
@@ -268,16 +268,49 @@ define void @foo({i32, i8} %v0) {
...
+  EXPECT_TRUE(VecTy->getElementType()->isIntegerTy(16));
+  EXPECT_EQ(VecTy->getElementCount(), ElementCount::getFixed(4));
+
+  auto *FVecTy = cast<sandboxir::VectorType>(F->getArg(1)->getType());
+  EXPECT_TRUE(FVecTy->getElementType()->isFloatTy());
+  auto *IVecTy = sandboxir::VectorType::getInteger(Ctx, FVecTy);
+  EXPECT_TRUE(IVecTy->getElementType()->isIntegerTy(32));
+  EXPECT_EQ(IVecTy->getElementCount(), FVecTy->getElementCount());
+
+  auto *ExtVecTy =
```
Comment (vporpo) [inline llvm/unittests/SandboxIR/TypesTest.cpp:289]: nit: I usually add a comment mentioning the function being tested. This makes the test a bit more readable, especially in cases where you a check spans multiple lines of code. In this case something like `// getExtendedVectorType`

File: llvm/include/llvm/SandboxIR/Type.h [hunk 3]
```diff
@@ -317,7 +317,32 @@ class StructType : public Type {
 class VectorType : public Type {
 public:
   static VectorType *get(Type *ElementType, ElementCount EC);
-  // TODO: add missing functions
+  static VectorType *get(Type *ElementType, unsigned NumElements,
+                         bool Scalable) {
+    return VectorType::get(ElementType,
+                           ElementCount::get(NumElements, Scalable));
+  }
+  // Needs tests
```
Comment (vporpo) [inline llvm/include/llvm/SandboxIR/Type.h:325]: Hmm which tests are missing?

Comment (Sterling-Augustine) [reply llvm/include/llvm/SandboxIR/Type.h:325]: Delete stale comment

File: llvm/unittests/SandboxIR/TypesTest.cpp (hunk 2 above)
Comment (Sterling-Augustine) [reply llvm/unittests/SandboxIR/TypesTest.cpp:289]: done

File: llvm/include/llvm/SandboxIR/Type.h (hunk 1 above)
Comment (Sterling-Augustine) [reply llvm/include/llvm/SandboxIR/Type.h:335]: done

File: llvm/unittests/SandboxIR/TypesTest.cpp [hunk 4]
```diff
@@ -268,16 +268,50 @@ define void @foo({i32, i8} %v0) {
...
   auto *F = Ctx.createFunction(LLVMF);
-  // Check classof(), creation.
-  [[maybe_unused]] auto *VecTy =
-      cast<sandboxir::VectorType>(F->getArg(0)->getType());
+  // Check classof(), creation, accessors
+  auto *VecTy = cast<sandboxir::VectorType>(F->getArg(0)->getType());
+  EXPECT_TRUE(VecTy->getElementType()->isIntegerTy(16));
+  EXPECT_EQ(VecTy->getElementCount(), ElementCount::getFixed(4));
+
+  auto *FVecTy = cast<sandboxir::VectorType>(F->getArg(1)->getType());
```
Comment (vporpo) [inline llvm/unittests/SandboxIR/TypesTest.cpp:283]: I think we are not testing the new `get()` functions:
```
// get(ElementType, NumElements, Scalable)
EXPECT_EQ(VectorType::get(Type::getInt16Ty(Ctx), 4, **/*Scalable=*/false), F>getArg(0)->getType());

// get(ElementType, Other)
EXPECT_EQ(VectorType::get(Type::getInt16Ty(Ctx), F->getArg(0)->getType()), F->getArg(0)->getType());

```

**Review by vporpo — APPROVED** [review]
LGTM, thanks!

**Pull request merged**